File: AoC/day1/counter.py

def counter1():
    fd = open("input.txt","r")
    depth = []
    incr = 0
    for line in fd:
        depth.append(int(line))
        if len(depth) == 1:
            continue
        if depth[-1] > depth[-2]:
            incr = incr+ 1
    print(incr)

def counter3():
    fd = open("input.txt","r")
    depth = []
    incr = 0
    for line in fd:
        depth.append(int(line))
        if len(depth) <= 3:
            continue
        if (depth[-1]+depth[-2]+depth[-3]) > (depth[-4]+depth[-2]+depth[-3]):
            incr = incr+ 1
    print(incr)

counter3()
    

File: AoC/day1/counter.java
import java.io.File;
import java.io.FileNotFoundException;
import java.util.ArrayList;
import java.util.Scanner;

public class counter{
    public static void counter1() throws FileNotFoundException{
        int count = 0;
        ArrayList<Integer> list = new ArrayList<Integer>();
        Scanner scan = new Scanner(new File("input.txt"));
        while (scan.hasNextLine()){
            String line = scan.nextLine();
            int num = Integer.parseInt(line);
            
            list.add(num);
            if (list.size() == 1) {
                ;
            }
            else if (num > list.get(list.size()-2)){
                count++;
            }
        }
        System.out.println(count);
    }

    
    public static void main(String[] args) throws FileNotFoundException{
        counter1();
    } 
}